src/core/Akka/Util/Result.cs

```diff
@@ -137,7 +137,30 @@ public static Result<T> Failure<T>(Exception exception)
         /// <returns>TBD</returns>
         public static Result<T> FromTask<T>(Task<T> task)
         {
-            return task.IsCanceled || task.IsFaulted ? new Result<T>(task.Exception) : new Result<T>(task.Result);
+            if(!task.IsCompleted)
+                throw new ArgumentException("Task is not completed. Result.FromTask only accepts completed tasks.", nameof(task));
+            
+            if(task.Exception is not null)
+                return new Result<T>(task.Exception);
+            
+            if (task.IsCanceled && task.Exception is null)
+            {
+                try
+                {
+                    _ = task.GetAwaiter().GetResult();
+                }
+                catch(Exception e)
+                {
+                    return new Result<T>(e);
+                }
+
+                throw new InvalidOperationException("Should never reach this line!");
+            }
```

Comment:
Fix, detect edge case where a `Task` is cancelled and its `Exception` property is null.

Reply:
LGTM